packages/code-studio/src/settings/SettingsUtils.tsx

```diff
@@ -0,0 +1,199 @@
+  columnType: string,
+  format: Partial<TableColumnFormat>
+): boolean {
+  // Undefined or empty string formats are always invalid
+  if (
+    !columnType ||
+    format.formatString == null ||
+    !isFormatStringFormat(format)
+  ) {
+    return false;
+  }
+  switch (columnType) {
+    case 'datetime':
+      return DateTimeColumnFormatter.isValid(format);
+    case 'decimal':
+      return DecimalColumnFormatter.isValid(format);
+    case 'int':
+      return IntegerColumnFormatter.isValid(format);
+    default: {
+      log.warn('Trying to validate format of unknown type');
+      return true;
+    }
+  }
+}
+
+export function removeFormatRuleExtraProps(
+  item: FormatterItem
+): Omit<FormatterItem, 'id' | 'isNewRule'> {
+  const { id, isNewRule, ...rest } = item;
+  return rest;
+}
+
+export function isFormatRuleValidForSave(rule: FormatterItem): boolean {
+  return (
+    isValidColumnName(rule.columnName) &&
+    isValidFormat(rule.columnType, rule.format)
+  );
+}
+
+export function renderTimeZoneOptions(): JSX.Element[] {
+  const options = TimeUtils.TIME_ZONES.map(timeZone => {
+    const { label, value } = timeZone;
+    return (
+      <option value={value} key={value}>
+        {label}
+      </option>
+    );
+  });
+
+  return options;
+}
+
+export function renderColumnTypeOptions(): ReactElement {
+  const columnTypesArray = [
+    { value: TableUtils.dataType.DATETIME, label: 'DateTime' },
+    { value: TableUtils.dataType.DECIMAL, label: 'Decimal' },
+    { value: TableUtils.dataType.INT, label: 'Integer' },
+  ];
+
+  const columnTypeOptions = columnTypesArray.map(item => (
+    <option key={`key-columnType-${item.value}`} value={item.value}>
+      {item.label}
+    </option>
+  ));
+
+  return (
+    <>
+      <option key="key-columnType-placeholder" disabled value="">
+        Select Type
+      </option>
+      {columnTypeOptions}
+    </>
+  );
+}
+
+export function renderDateTimeOptions(
+  timestamp: Date,
+  timeZone: string,
+  showTimeZone: boolean,
+  showTSeparator: boolean,
+  isGlobalOptions: boolean,
+  legacyGlobalFormat?: string
+): ReactElement[] {
+  const formatter = new Formatter([], {
+    timeZone,
+    showTimeZone,
+    showTSeparator,
+  });
+  const formats = isGlobalOptions
+    ? DateTimeColumnFormatter.getGlobalFormats(showTimeZone, showTSeparator)
+    : DateTimeColumnFormatter.getFormats(showTimeZone, showTSeparator);
+
+  if (legacyGlobalFormat != null && !formats.includes(legacyGlobalFormat)) {
+    formats.unshift(legacyGlobalFormat);
+  }
+
+  return formats.map(formatString => {
+    const format = DateTimeColumnFormatter.makeFormat(
+      '',
+      formatString,
+      DateTimeColumnFormatter.TYPE_GLOBAL
+    );
+    return (
+      <option value={formatString} key={formatString}>
+        {formatter.getFormattedString(
+          timestamp,
+          TableUtils.dataType.DATETIME,
+          '',
+          format
+        )}
+      </option>
+    );
+  });
+}
```

Comment:
These static functions are essentially just functional React components, they could be put into their own files, e.g. `TimeZoneOptions.tsx`, `ColumnTypeOptions.tsx`, and `DateTimeOptions.tsx`. Not a big deal though.